elastio-s3-changelog/cloudformation-single.yaml

```
@@ -0,0 +1,263 @@
+# This template enables the S3 changelog for a single bucket.
+AWSTemplateFormatVersion: "2010-09-09"
+Parameters:
+  BucketName:
+    Type: String
+    AllowedPattern: .{3,}
+    ConstraintDescription: Bucket name must be at least 3 characters long.
+    Description: S3 bucket name.
+
+  ScanExistingObjects:
+    Type: String
+    AllowedValues: ['true', 'false']
+    Default: 'false'
+    Description: >
+      If set to true, the full initial scan of the bucket will be performed.
+      If set to false, only new objects in the bucket will be scanned.
+
+  EnableQrts:
+    Type: String
+    AllowedValues: ['true', 'false']
+    Default: 'false'
+    Description: >
+      If set to true, enables quasi-realtime scanning.
+      If set to false, disables quasi-realtime scanning.
+
+  KeyPrefixes:
+    # Unfortunately, it's impossible to pass CommaDelimitedList to a nested stack,
+    # so we have to use String here, and then split it manually.
+    Type: String
+    AllowedPattern: ^(([^/,\s]+/)+)(,([^/,\s]+/)+)*$|^$
+    ConstraintDescription: Prefixes must end (but not start) with '/', and cannot contain spaces.
+    Description: >
+      Comma-delimited list of prefixes of objects to scan. If empty, all objects will be scanned.
+      Example: xyz/,foo/bar/
+
+Conditions:
+  ScanExistingObjects: !Equals [!Ref ScanExistingObjects, 'true']
+  EnableQrts: !Equals [!Ref EnableQrts, 'true']
+  NoKeyPrefixes: !Equals [!Ref KeyPrefixes, '']
+
+Resources:
+  queue:
+    Type: AWS::SQS::Queue
+    Properties:
+      MessageRetentionPeriod: 1209600 # 14 days
+      VisibilityTimeout: 43200 # 12 hours
+      RedrivePolicy:
+        deadLetterTargetArn: !GetAtt dlq.Arn
+        maxReceiveCount: 3
+      Tags:
+        - Key: elastio:resource
+          Value: 'true'
+        - Key: elastio:s3-changelog-bucket
+          Value: !Ref BucketName
+        - Key: elastio:s3-changelog-stack-arn
+          Value: !Ref AWS::StackId
+
+  dlq:
+    Type: AWS::SQS::Queue
+    Properties:
+      MessageRetentionPeriod: 1209600
+      VisibilityTimeout: 43200
+      Tags:
+        - Key: elastio:resource
+          Value: 'true'
+        - Key: elastio:s3-changelog-dlq
+          Value: !Ref BucketName
+        - Key: elastio:s3-changelog-stack-arn
+          Value: !Ref AWS::StackId
+
+  eventRule:
+    Type: AWS::Events::Rule
+    DependsOn: queue
+    Properties:
+      Description: !Sub Track new objects creation in ${BucketName}
```

Let's keep the description more generic. Maybe we'll subscribe to more events in the future.
```suggestion
      Description: !Sub Track S3 object change events in ${BucketName}
```